src/reducers/todoReducers.js

```diff
@@ -1,32 +1,54 @@
-import { ADD_TODO, DELETE_TODO } from "actions/actionTypes";
+import { ADD_TODO, DELETE_TODO, COMPLETE_TODO } from "actions/actionTypes";
 const initialData = {
   list: [],
 };
 
 const todoReducers = (state = initialData, action) => {
   switch (action.type) {
-    case ADD_TODO:
-      const { id, task, date } = action.payload;
+    case ADD_TODO: {
+      const { id, task, date, isTaskComplete, completeTime } = action.payload;
       return {
         ...state,
         list: [
           {
             id,
             task,
             date,
+            isTaskComplete,
+            completeTime,
           },
           ...state.list,
         ],
       };
+    }
 
-    case DELETE_TODO:
+    case DELETE_TODO: {
       const newList = state.list.filter(
         (task) => task.id !== action.payload.id
       );
       return {
         ...state,
         list: newList,
       };
+    }
+
+    case COMPLETE_TODO: {
+      const { id, completeTime, isTaskComplete } = action.payload;
+      const completedTaskIndex = state.list.findIndex((task) => task.id === id);
+
+      const completedTask = {
+        ...state.list[completedTaskIndex],
+        completeTime,
+        isTaskComplete,
+      };
+      const completedList = state.list.map((todo) => ({ ...todo }));
+      completedList[completedTaskIndex] = completedTask;
```

Comment:
index operation is redundant, consider the following simplification.

```suggestion
      const completedList = state.list.map((todo) => {
          if (task.id === id) {
              return {...todo, completeTime, isTaskComplete, }
          }

          return todo
      });
```